pkg/k8s/fake.go

```diff
@@ -19,29 +22,64 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
-// NewFakeClientSets provides a mock Kubernetes ClientSet.
-func NewFakeClientSets(configs ...string) (kubernetes.Interface, spclient.Interface, error) {
+// NewFakeAPI provides a mock KubernetesAPI backed by hard-coded resources
+func NewFakeAPI(configs ...string) (*KubernetesAPI, error) {
+	client, apiextClient, _, err := NewFakeClientSets(configs...)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KubernetesAPI{
+		Interface:     client,
+		Apiextensions: apiextClient,
+	}, nil
+}
+
+// NewFakeAPIFromManifests reads from a slice of readers, each representing a
+// manifest or collection of manifests, and returns a mock KubernetesAPI.
+func NewFakeAPIFromManifests(readers []io.Reader) (*KubernetesAPI, error) {
+	client, apiextClient, _, err := newFakeClientSetsFromManifests(readers)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KubernetesAPI{
+		Interface:     client,
+		Apiextensions: apiextClient,
+	}, nil
+}
+
+// NewFakeClientSets provides mock Kubernetes ClientSets.
+// TODO: make this private once KubernetesAPI (and NewFakeAPI) supports spClient
+func NewFakeClientSets(configs ...string) (kubernetes.Interface, apiextensionsclient.Interface, spclient.Interface, error) {
```

Comment:
What are your thoughts on making `NewFakeClientSets` return `kubernetesAPI` instead of the two clientsets? Are there instances where we would want to just use the `NewFakeClientSets` instead of `NewFakeAPI`.

Reply:
I agree replacing `kubernetes.Interface` and `apiextensionsclient.Interface` with a single `kubernetesAPI` is a good intermediate step. I'm going to leave this for now, because this function still needs to return an `spclient.Interface`, since `kubernetesAPI` does not yet support it. Once we do (via #2751), we can make this function private, and everyone can call `NewFakeAPI`.